Sources/Classes/Views/BigBossButton/BigBossButtonView.swift

```diff
@@ -0,0 +1,379 @@
+    }
+
+    // MARK: - Stored Properties
+
+    public var spinnerView: Spinner? {
+        willSet {
+            if newValue == nil {
+                removeSpinner()
+            }
+        }
+
+        didSet {
+            if spinnerView != nil {
+                addSpinner()
+            }
+        }
+    }
+
+    private let button = BigBossButton()
+
+    public var shadowView = UIView() {
+        willSet {
+            shadowView.removeFromSuperview()
+        }
+        didSet {
+            insertSubview(shadowView, at: 0)
+            configureShadowViewConstraints()
+        }
+    }
+
+    private var isEnabledObserver: NSKeyValueObservation?
+    private var isHighlightedObserver: NSKeyValueObservation?
+
+    public var tapOnDisabledButton: VoidBlock?
+
+    public var appearance = Appearance() {
+        didSet {
+            configureAppearance()
+            configureConstraints()
+        }
+    }
+
+    // MARK: - Initialization
+
+    override public init(frame: CGRect) {
+        super.init(frame: frame)
+        initializeView()
+    }
+
+    required public init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        initializeView()
+    }
+
+    // MARK: - Drivers
+
+    open var tapObservable: Observable<Void> {
+        return button.rx.tap.asObservable()
+    }
+
+    // MARK: - Button State Observation
+
+    //    private func observeIsEnabled() -> NSKeyValueObservation {
+    //        return button.observe(\BigBossButton.isEnabled, options: .new) { [weak self] _, isEnabled in
+    //
+    //            guard let self = self else {
+    //                return
+    //            }
+    //
+    //            if let isEnabled = isEnabled.newValue {
+    //                var state = self.stateRelay.value
+    //                state.subtract([.enabled, .disabled])
+    //                state.insert(isEnabled ? .enabled : .disabled)
+    //                self.stateRelay.accept(state)
+    //            }
+    //        }
+    //    }
+    //
+    //    private func observeIsHighlighted() -> NSKeyValueObservation {
+    //        return button.observe(\BigBossButton.isHighlighted, options: .new,
+    //      changeHandler: { [weak self] _, isHighlighted in
+    //
+    //            guard let self = self else {
+    //                return
+    //            }
+    //
+    //            if let isHighlighted = isHighlighted.newValue {
+    //                var state = self.stateRelay.value
+    //                state.subtract([.highlighted, .unhighlighted])
+    //                state.insert(isHighlighted ? .highlighted : .unhighlighted)
+    //                self.stateRelay.accept(state)
+    //            }
+    //        })
+    //    }
+
+    // MARK: - UI
+
+    override open func layoutSubviews() {
+        super.layoutSubviews()
+        shadowView.layer.shadowPath = UIBezierPath(rect: button.bounds).cgPath
+    }
+
+    public var buttonIsDisabledWhileLoading = false
```

Comment:
group vars with vars and methods with methods or group by access modifiers

Reply:
fixed